osquery/filesystem/filesystem.cpp

```diff
@@ -438,6 +438,44 @@ Status isDirectory(const fs::path& path) {
   return Status(ec.value(), ec.message());
 }
 
+Status createDirectory(const boost::filesystem::path& dir_path,
+                       bool const recursive,
+                       bool const ignore_existence) {
+  auto err = boost::system::error_code{};
+  bool is_created = false;
+  if (recursive) {
+    is_created = boost::filesystem::create_directories(dir_path, err);
+  } else {
+    is_created = boost::filesystem::create_directory(dir_path, err);
+  }
+  if (is_created) {
+    return Status::success();
+  }
+  if (ignore_existence && isDirectory(dir_path).ok()) {
+    return Status::success();
+  }
+  auto msg = std::string{"Could not create directory "} + dir_path.string();
+  if (err) {
+    msg += ": ";
+    msg += err.message();
+  }
+  return Status::failure(msg);
+}
+
+Status createDirectoryRecursive(const boost::filesystem::path& dir_path) {
```

Comment:
Why do we need separate implementation for that? createDirectory can do it anyway, if it's just for simplicity why don't we  call something like createDirectory(dir_path, true, true);

Reply:
Oh, thank you! I just forgot to delete it.